metric/cpu/metrics.go

```diff
@@ -84,6 +84,18 @@ The below code implements a "metrics tracker" that gives us the ability to
 calculate CPU percentages, as we average usage across a time period.
 */
 
+type option struct {
+	usePerformanceCounter bool
+}
+
+type OptionFunc func(*option)
+
+func WithPerformanceCounter() OptionFunc {
```

Comment:
I assume performance counters is just a windows thing? If so, we could add a comment at the top of this function that states that it only applies to windows.

Reply:
Putting Windows in the option name would be even more obvious.

Also, consider that we are probably going to want to make this the default once we are satisfied with it. Probably just deleting the option would work once we do that but something to consider for the near future.